core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -118,9 +119,14 @@ class BrokerMetadataListener(
       }
       _publisher.foreach(publish)
 
+      // If we detected a change in metadata.version, generate a local snapshot
+      val metadataVersionChanged = Option(_delta.featuresDelta()).exists { featuresDelta =>
```

Comment:
It would be good to add an accessor for MetadataDelta like `MetadataDelta#metadataVersionChanged` (which I suppose could return an `Optional<MetadataVersion>` for simplicity)